src/server/passive_tran_server.cpp

```diff
@@ -35,3 +57,84 @@ passive_tran_server::on_boot ()
 {
   assert (is_passive_transaction_server ());
 }
+
+tran_server::request_handlers_map_t
+passive_tran_server::get_request_handlers ()
+{
+  request_handlers_map_t::value_type log_header_log_append_prev_lsa_handler_value =
+	  std::make_pair (page_to_tran_request::SEND_LOG_HEADER_LOG_APPEND_PREV_LSA,
+			  std::bind (&passive_tran_server::receive_log_header_log_append_prev_lsa,
+				     std::ref (*this), std::placeholders::_1));
+
+  std::map<page_to_tran_request, std::function<void (cubpacking::unpacker &upk)>> handlers_map =
+	      tran_server::get_request_handlers ();
+
+  handlers_map.insert (log_header_log_append_prev_lsa_handler_value);
+
+  return handlers_map;
+}
+
+void
+passive_tran_server::receive_log_header_log_append_prev_lsa (cubpacking::unpacker &upk)
+{
+  std::string message;
+  upk.unpack_string (message);
+
+  const int log_page_size = db_log_page_size ();
+  assert (message.size () == (2 * log_page_size + sizeof (struct log_lsa)));
+
+  // pass to caller thread; it has the thread context needed to access engine functions
+  {
+    std::lock_guard<std::mutex> lockg { m_log_header_log_append_prev_lsa_mtx };
+    m_log_header_log_append_prev_lsa.swap (message);
+  }
+  m_log_header_log_append_prev_lsa_condvar.notify_one ();
+}
+
+void passive_tran_server::send_and_receive_log_header_log_append_prev_lsa (THREAD_ENTRY *thread_p)
+{
+  assert (m_log_header_log_append_prev_lsa.empty ());
+
+  // empty message request
+  push_request (tran_to_page_request::SEND_LOG_HEADER_LOG_APPEND_PREV_LSA_FETCH, std::string ());
+
+  {
+    std::unique_lock<std::mutex> ulock { m_log_header_log_append_prev_lsa_mtx };
+    // TODO: wait_for a limited time in case page server hangs
+    m_log_header_log_append_prev_lsa_condvar.wait (ulock, [this] ()
+    {
+      return !m_log_header_log_append_prev_lsa.empty ();
+    });
+  }
+
+  const char *message_buf = m_log_header_log_append_prev_lsa.c_str ();
+  const int log_page_size = db_log_page_size ();
+
+  // log header, copy and initialize header
+  assert (log_Gl.loghdr_pgptr != nullptr);
+  std::memcpy (reinterpret_cast<char *> (log_Gl.loghdr_pgptr), message_buf, log_page_size);
+  LOG_HEADER *const log_hdr = reinterpret_cast<LOG_HEADER *> (log_Gl.loghdr_pgptr->area);
+  log_Gl.hdr = *log_hdr;
+  message_buf += log_page_size;
+
+  // log append
+  assert (log_Gl.append.log_pgptr == nullptr);
+  /* fetch the append page; append pages are always new pages */
+  log_Gl.append.log_pgptr = logpb_create_page (thread_p, log_Gl.hdr.append_lsa.pageid);
```

Comment:
`logpb_create_page`, internally, has logic to flush a log page in case the round robin mechanism wants to re-use/overwrite an existing buffer. Given the context, that part should never be hit. Best would be to add an argument to ensure no flush ever occurs.

Reply:
No need to do it now.